1.3 release notes (#3880)
elastic/cloud-on-k8s: docs/release-notes/1.3.0.asciidoc

[[release-notes-1.3.0]]
== {n} version 1.3.0

[[breaking-1.3.0]]
[float]
=== Breaking changes

* Enforce a strict upgrade order between stack components {pull}3537[#3537] (issues: {issue}2600[#2600], {issue}3535[#3535])
* Add set-default-security-context flag to the operator {pull}3342[#3342] (issue: {issue}2791[#2791])

[[feature-1.3.0]]
[float]
=== New features

* Make timeouts configurable {pull}3782[#3782] (issue: {issue}684[#684])
* Support Elasticsearch volumes expansion {pull}3752[#3752] (issues: {issue}325[#325], {issue}3757[#3757], {issue}3767[#3767])
* Add DeploymentStrategy to Beats CRDs {pull}3633[#3633] (issue: {issue}3485[#3485])
* Add operator election support {pull}3632[#3632] (issue: {issue}709[#709])

[[enhancement-1.3.0]]
[float]
=== Enhancements

* Add license info to telemetry {pull}3859[#3859]
* Validate whether the storage class allows volume expansion before resizing volumes {pull}3795[#3795] (issue: {issue}3767[#3767])
* Add external IP addresses to certificate {pull}3791[#3791] (issue: {issue}910[#910])
* Support a UBI-only mode in the operator {pull}3789[#3789]
* Add CRD usage reporting {pull}3784[#3784]

> * Refactor Helm chart {pull}3711[#3711]

Reviewer: Shall I re-title this PR to say "Refactor Helm chart for public usage" and re-label it as a feature?

Author: I went ahead and did that already.

> * Refactor node config and add role validation {pull}3694[#3694] (issue: {issue}3409[#3409])

Reviewer 2: ```suggestion
* Refactor node configuration and add role validation {pull}3694[#3694] (issue: {issue}3409[#3409])
```

* Add support for configuration changes under OLM  {pull}3639[#3639] (issue: {issue}3591[#3591])

> * Remove memory leak mitigation {pull}3585[#3585]

Reviewer: Maybe re-title to say "Remove unnecessary memory leak mitigation patch"

* Add a container suffix flag to the operator {pull}3582[#3582] (issue: {issue}3552[#3552])
* Allow the operator to be configured from a file {pull}3570[#3570] (issue: {issue}3401[#3401])
* Reflect currently running version from status.version in additionalPrinterColumns {pull}3549[#3549]
* Report the lowest running version in the status of each resource {pull}3489[#3489] (issue: {issue}2600[#2600])
* Add WebhookConfigurationName Flag (#3424) {pull}3425[#3425]

> * Update module sigs.k8s.io/controller-runtime to v0.6.1 {pull}3417[#3417]

Reviewer: I doubt this is very useful for end users to know.

Author: I removed all others, missed this one, thanks.

* Add metadata for transform node role  {pull}3411[#3411]
* Add support for the node.roles setting {pull}3387[#3387] (issue: {issue}3324[#3324])
* Add node shard allocation awareness by default {pull}3380[#3380] (issue: {issue}2827[#2827])
* Add metadata for transform node role {pull}3361[#3361]
* Allow init containers modifications {pull}3336[#3336] (issue: {issue}2306[#2306])

[[bug-1.3.0]]
[float]
=== Bug fixes

> * Add role namespace when generating rbac for managed namespaces in helm charts {pull}3792[#3792]

Reviewer 2: ```suggestion
* Add role namespace when generating RBAC for managed namespaces in helm charts {pull}3792[#3792]
```

Reviewer: It doesn't really need to be in here because it's not a bug fix to something out in the wild. I added the exclude label to it.

> * Support new auth config format starting EnterpriseSearch 8.x {pull}3728[#3728] (issue: {issue}3725[#3725])

Reviewer 2: ```suggestion
* Support new auth config format starting EnterpriseSearch 8.x {pull}3728[#3728] (issue: {issue}3725[#3725])
```
=> Could it be _authoring configuration format_?

:issue: https://github.com/elastic/cloud-on-k8s/issues/
:pull: https://github.com/elastic/cloud-on-k8s/pull/

[[release-notes-1.3.0]]
== {n} version 1.3.0

[[breaking-1.3.0]]
[float]
=== Breaking changes

* Make ECK IPv6 compatible {pull}3654[#3654] (issues: {issue}2830[#2830], {issue}3649[#3649]) - in an accompanying change `network.publish_host` setting value was switched to DNS name. This might result in non-ready Pods being included in the headless service for Elasticsearch StatefulSets.

> * Enforce a strict upgrade order between stack components {pull}3537[#3537] (issues: {issue}2600[#2600], {issue}3535[#3535]) - ECK will prevent updates that would violate link:https://www.elastic.co/guide/en/elastic-stack/current/upgrading-elastic-stack.html#upgrade-order-elastic-stack[Elastic Stack upgrade process] version constraints. ECK 1.3.0 won't apply any updates to resources that are in violation of those constraints until versions are update to (at least) the same minors.

Reviewer 3: ```suggestion
* Enforce a strict upgrade order between stack components {pull}3537[#3537] (issues: {issue}2600[#2600], {issue}3535[#3535]) - ECK will prevent updates that would violate link:https://www.elastic.co/guide/en/elastic-stack/current/upgrading-elastic-stack.html#upgrade-order-elastic-stack[Elastic Stack upgrade process] version constraints. ECK 1.3.0 won't apply any updates to resources that are in violation of those constraints until versions are updated to (at least) the same minors.
```

> * Add set-default-security-context flag to handle runAs user in ES 8.0+ {pull}3342[#3342] (issue: {issue}2791[#2791]) - by default, a default security context will be added to Elasticsearch 8.0+ Pods.

Reviewer 3: ```suggestion
* Add set-default-security-context flag to handle runAs user in ES 8.0+ {pull}3342[#3342] (issue: {issue}2791[#2791]) - a default security context is added to Elasticsearch 8.0+ Pods in order to deal with volume filesystem permissions. Can be disabled through operator configuration on environments with restricted security settings (Openshift, for example).
```
